### ISSUE TITLE
[FEATURE] Allow more information to be specified about the target of a file

### ISSUE THREAD

**Issue by SuperCuber** [SuperCuber/dotter#18]
**Is your feature request related to a problem? Please describe.**
I want to be able to perform operations other than overwriting a file or variable in `local.toml`, for example appending certain text to it for machine-specific bashrc configuration for example.
Another use-case would be to specify the type of the file in `global.toml`.

**Describe the solution you'd like**
```toml
[package.files]
my_file = {}
```
A file mapped to an object with some description of the action wanted.\
An idea off the top of my head: `{"action": "append", "text": "echo hello world!"}`

**Describe alternatives you've considered**
The current workaround for this is to implement that behavior myself in the source file in an `{{if}}` block and then toggle that on in the machine's `local.toml` and off in all other `local.toml`s.

- [x] Update documentation when this is implemented
- [x] Think about how merging two target objects should work
    `local.toml` will just overwrite the whole value currently. I thought about making it merge in certain cases (for example specifying `append` in global and `prepend` in local) but this seems like too narrow of a use-case to matter.
- [ ] When this is done, open an issue for this to also work on folders (?)

**Comment by SuperCuber** [comment]
More thought out example:
```toml
[package.files.source_file]
target = "~/.target_file"
append = """
echo hello world!
"""
prepend = """
PATH=$PATH:~/bin
"""
type = "symbolic"  # or "template"
```

**Comment by mnivoliez** [comment]
Could this syntax also work:

```toml
[package.files]
source_file = { target: "~/.target_file", type = "symbolic"}
```

or even:
```toml
[package.files]
source_dir = { target: "~/.target_dir", type = "symbolic"}
```

**Comment by SuperCuber** [comment]
@mnivoliez these are actually equivalent in terms of toml, you can even mix and match them:

```toml
[package]
variables = {"foo": "bar"}

[package.files]
a = "b"
```
Is a valid configuration.

If you mean file vs dir, yes I need to make sure these are expanded correctly for directories - for example, how would `append` on a directory be handled? Probably an error but gotta make sure

I considered symlinking a whole directory in the past but it complicates the implementation and explanation a lot (what if they add a templated file inside the directory?) for not a lot of gain.

**Comment by mnivoliez** [comment]
@SuperCuber The current system is good (symlinking or copying files in dir instead of symlinking the dir). But the the dir should be able to be flagged as symlink or template.

**Comment by SuperCuber** [comment]
@mnivoliez I see, and in that case you'd expect it to forcefully override all the files inside that directory no matter if they look like symlink or template, right?

I'd still need to make an exception for non-utf8 files for technical reasons which might be slightly confusing...\
I think for next version's implementation I'll disallow the whole extended configuration on directories since it's for pretty exceptional cases, and then add that as an [enhancement] issue for later

**Comment by SuperCuber** [comment]
Implemented in f4f241481fdb910ad25d592ad3da83ca5aceecf3

**Comment by SuperCuber** [comment]
@mnivoliez This has been implemented for files, check out the relevant [new section](https://github.com/SuperCuber/dotter/wiki/Setup-and-Configuration#complex-target) in the wiki.
